artichoke-backend/src/extn/core/matchdata/mod.rs

```diff
@@ -308,7 +308,12 @@ impl MatchData {
             } else {
                 haystack.len()
             };
-            let offset = self.region.offset();
+            let offset = self
+                .haystack
+                .char_indices()
```

Comment:
calling `bstr` directly here is buggy for a couple of reasons:

- It presumes that the `String` value the haystack bytes were derived from has a UTF-8 string encoding.
- `bstr`'s `CharIndices` iterator, like all of the library's UTF-8 oriented methods, has different behavior than Artichoke for some types of invalid UTF-8 byte sequences. `bstr` uses the [_substitution of maximal subparts_ replacement strategy](https://docs.rs/bstr/latest/bstr/#handling-of-invalid-utf-8) which means some invalid UTF-8 byte sequences like `\xF0\x9F\x8` get collapsed to one replacement character, whereas Artichoke treats these 3 bytes as 3 characters.